Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogHeaderWidget.cpp

```diff
@@ -187,82 +194,126 @@ namespace O3DE::ProjectManager
+    }
 
-                for (int downloadingGemNumber = 0; downloadingGemNumber < downloadQueue.size(); ++downloadingGemNumber)
-                {
-                    QHBoxLayout* nameProgressLayout = new QHBoxLayout();
-                    TagWidget* newTag = new TagWidget(downloadQueue[downloadingGemNumber]);
-                    nameProgressLayout->addWidget(newTag);
-                    QLabel* progress = new QLabel(downloadingGemNumber == 0? QString("%1%").arg(downloadProgress) : tr("Queued"));
-                    nameProgressLayout->addWidget(progress);
-                    QSpacerItem* spacer = new QSpacerItem(0, 0, QSizePolicy::Expanding, QSizePolicy::Minimum);
-                    nameProgressLayout->addSpacerItem(spacer);
-                    QLabel* cancelText = new QLabel(QString("<a href=\"%1\">Cancel</a>").arg(downloadQueue[downloadingGemNumber]));
-                    cancelText->setTextInteractionFlags(Qt::LinksAccessibleByMouse);
-                    connect(cancelText, &QLabel::linkActivated, this, &CartOverlayWidget::OnCancelDownloadActivated);
-                    nameProgressLayout->addWidget(cancelText);
-                    downloadingItemLayout->addLayout(nameProgressLayout);
-                    QProgressBar* downloadProgessBar = new QProgressBar();
-                    downloadingItemLayout->addWidget(downloadProgessBar);
-                    downloadProgessBar->setValue(downloadingGemNumber == 0 ? downloadProgress : 0);
-                }
+    void CartOverlayWidget::GemDownloadRemoved(const QString& gemName)
+    {
+        QWidget* gemToRemove = m_downloadingListWidget->findChild<QWidget*>(gemName);
+        QLayout* gemToRemoveLayout = gemToRemove ? gemToRemove->layout() : nullptr;
 
-                widget->setUpdatesEnabled(true);
-                widget->show();
+        if (gemToRemoveLayout)
+        {
+            QLayoutItem* rowLayoutItem = nullptr;
+            while ((rowLayoutItem = gemToRemoveLayout->layout()->takeAt(0)) != nullptr)
+            {
+                rowLayoutItem->widget()->deleteLater();
             }
-        };
+            gemToRemoveLayout->layout()->deleteLater();
+            gemToRemove->deleteLater();
+        }
 
-        auto downloadEnded = [=](bool /*success*/)
+        const AZStd::vector<QString>& downloadQueue = m_downloadController->GetDownloadQueue();
+
+        if (m_downloadController->IsDownloadQueueEmpty())
         {
-            update(0); // update the list to remove the gem that has finished
-        };
-        // connect to download controller data changed
-        connect(m_downloadController, &DownloadController::GemDownloadProgress, this, update);
-        connect(m_downloadController, &DownloadController::Done, this, downloadEnded);
-        update(0);
+            m_downloadSectionWidget->hide();
+        }
+        else
+        {
+            QLabel* numDownloads = m_downloadingListWidget->findChild<QLabel*>("NumDownloadsInProgressLabel");
```

Comment:
Should move downloadQueue declaration to here. Or even better declare downloadQueueSize then use it below.